runtime/protocol/http/common/src/aws/smithy/kotlin/runtime/http/endpoints/Endpoint.kt

```diff
@@ -27,9 +35,18 @@ import aws.smithy.kotlin.runtime.http.Url
  * is expected to be mutable and the client cannot modify the endpoint correctly, the operation
  * will likely fail.
  */
-public data class Endpoint(
+public data class Endpoint @InternalApi constructor(
```

Comment:
why is this constructor internal and why do we still need the constructor on L47?

Reply:
Callers (people writing endpoint resolvers of their own) aren't allowed to directly manipulate the properties field. This is SEP-dictated. There's been a LOT of discussion around this field internally, we can discuss further.